test/integration/context-based-restrictions.v1.test.js

```diff
@@ -0,0 +1,837 @@
+  });
+
+  test('replaceZone() - 412 - invalid IfMath parameter', async () => {
+    // The zone could not be updated due to invalid IfMath parameter.
+    // Request models needed by this operation.
+
+    // AddressIPAddress
+    const addressModel = {
+      type: 'ipAddress',
+      value: '169.24.57.235',
+    };
+
+    const params = {
+      zoneId,
+      ifMatch: 'abc',
+      name: 'an example of invalid IfMath parameter',
+      accountId,
+      addresses: [addressModel],
+      description: 'this is an example of invalid IfMath parameter',
+      transactionId: uuidv4(),
+    };
+
+    await expect(contextBasedRestrictionsService.replaceZone(params)).rejects.toMatchObject({
+      status: 412,
+    });
+  });
+
+  test('listAvailableServicerefTargets() - 200', async () => {
+    const params = {
+      type: 'all',
+      transactionId: uuidv4(),
+    };
+
+    const res = await contextBasedRestrictionsService.listAvailableServicerefTargets(params);
+    expect(res).toBeDefined();
+    expect(res.status).toBe(200);
+    expect(res.result).toBeDefined();
+  });
+
+  test('listAvailableServicerefTargets() - 400 - invalid type', async () => {
+    // type value is not supported
+    const params = {
+      type: 'invalid-type',
+      transactionId: uuidv4(),
+    };
+
+    await expect(
+      contextBasedRestrictionsService.listAvailableServicerefTargets(params)
+    ).rejects.toMatchObject({
+      status: 400,
+    });
+  });
+
+  test('createRule() - 201', async (done) => {
+    // Request models needed by this operation.
+
+    // RuleContextAttribute
+    const ruleContextAttributeModel = {
+      name: 'networkZoneId',
+      value: zoneId,
+    };
+
+    // RuleContext
+    const ruleContextModel = {
+      attributes: [ruleContextAttributeModel],
+    };
+
+    // ResourceAttribute
+    const resourceAttributeAccountIdModel = {
+      name: 'accountId',
+      value: accountId,
+      operator: 'stringEquals',
+    };
+
+    const resourceAttributeServiceNameModel = {
+      name: 'serviceName',
+      value: serviceName,
+      operator: 'stringEquals',
+    };
+
+    // ResourceTagAttribute
+    const resourceTagAttributeModel = {
+      name: 'aTagName',
+      value: 'aTagValue',
+      operator: 'stringEquals',
+    };
+
+    // Resource
+    const resourceModel = {
+      attributes: [resourceAttributeAccountIdModel, resourceAttributeServiceNameModel],
+      tags: [resourceTagAttributeModel],
+    };
+
+    const params = {
+      contexts: [ruleContextModel],
+      resources: [resourceModel],
+      description: 'this is an example of rule',
+      transactionId: uuidv4(),
+    };
+
+    let res;
+    try {
+      res = await contextBasedRestrictionsService.createRule(params);
+    } catch (err) {
+      done(err);
+    }
+
+    expect(res).toBeDefined();
+    expect(res.status).toBe(201);
+    expect(res.result).toBeDefined();
+
+    ruleId = res.result.id;
+    ruleEtag = res.headers.etag;
+
+    done();
+  });
+
```

Comment:
I'd like to ask you to add a test case which covers status code `400` case. The other 3 sdks have a test for this case and it is called `CreateRule - Create a rule with 'service not cbr enabled' error` in the Go for example.

Reply:
added